Source/Fuse.Controls.Native/Android/SystemScreenConfigAndroidImpl.uno

```diff
@@ -0,0 +1,104 @@
+using Fuse.Controls.Native;
+
+namespace Fuse.Controls.Native
+{
+	extern(Android) internal class SystemScreenConfigAndroidImpl
+	{
+		private static SystemUiVisibility.Flag _wantedFlags = SystemUiVisibility.Flag.None;
+		private static SystemScreenConfig config;
+
+		public static void Init(SystemScreenConfig config)
+		{
+			SystemScreenConfigAndroidImpl.config = config;
+			SystemUiVisibility.VisibilityChanged += VisibilityChanged;
+		}
+
+		public static void DeInit()
+		{
+			SystemUiVisibility.VisibilityChanged -= VisibilityChanged;
+			SystemScreenConfigAndroidImpl.config = null;
+		}
+
+		public static void TimerDone()
+		{
+			SystemUiVisibility.Flags = _wantedFlags;
+		}
+
+		public static void SetShowState(SystemScreenConfig.Visibility state)
+		{
+			switch(state)
+			{
+				case SystemScreenConfig.Visibility.None:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.Fullscreen | SystemUiVisibility.Flag.HideNavigation;
+					_wantedFlags = SystemUiVisibility.Flag.Fullscreen | SystemUiVisibility.Flag.HideNavigation;
+				break;
+				case SystemScreenConfig.Visibility.Minimal:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.HideNavigation;
+					_wantedFlags = SystemUiVisibility.Flag.HideNavigation;
+				break;
+				case SystemScreenConfig.Visibility.Full:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.None;
+					_wantedFlags = SystemUiVisibility.Flag.None;
+				break;
+			}
+		}
+
+		public static void SetStatusState(bool val)
+		{
+			if(val)
+			{
+				_wantedFlags = _wantedFlags ^ SystemUiVisibility.Flag.Fullscreen;
+				SystemUiVisibility.Flags = SystemUiVisibility.Flags ^ SystemUiVisibility.Flag.Fullscreen;
```

Comment:
Not neccesarily, as the Flags might contain other flags this code doesn't deal with yet. I have written the code with the fact that not all the flags are being controlled by the code in mind.
Setting Flags to _wantedFlags directly would therefore step on potential other code controlling other flags that this code does not know of, so i feel this is safest as long as not all the flags are controlled from here.